sourcegraph/sourcegraph
# Move git-extras functionality to the core workflow

Closes https://github.com/sourcegraph/sourcegraph/issues/39433

1. Migrates [git-extras extension](https://sourcegraph.com/extensions/sourcegraph/git-extras) functionality to the core workflow (available under `extensionsAsCoreFeatures` experimental feature). 
This PR migrates only blame for the whole file mode and omits blame for selected lines mode. Whether we migrate the latter or not depends on the updates to the git blame UI (see [this comment](https://github.com/sourcegraph/sourcegraph/issues/38377#issuecomment-1193655493)) and will be addressed (if needed) in the follow-up PRs.
In the file view, blame decorations are displayed only in the column view (see https://github.com/sourcegraph/sourcegraph/pull/36007).
The only UI change is that git blame icon migrated from the action items bar (extensions sidebar) to the repo header section (see screenshots).
2. Makes blame visibility status unique to the page (blame decorations are hidden initially) as opposed to the user setting synced across all the pages (inspired by how blame works in the VSCode for example).

[Loom](https://www.loom.com/share/ba02f360a9ee4afe8793cf215c691e90)

<details>
<summary>Screenshots</summary>
<table>
<tbody>
<tr>
<td>File view</td>
<td>

<img width="2555" alt="Screenshot 2022-07-26 at 15 21 45" src="https://user-images.githubusercontent.com/25318659/181005218-a06b0274-4e50-42b9-a2c1-d2caee6efd90.png">


</td>
</tr>
<tr>
<td>Commit view</td>
<td>

<img width="2555" alt="Screenshot 2022-07-26 at 15 22 44" src="https://user-images.githubusercontent.com/25318659/181005099-4c176d24-a9e5-4c64-808c-05995425ee9c.png">

</td>
</tr>
<tr>
<td>Compare view</td>
<td>
<img width="2555" alt="Screenshot 2022-07-26 at 15 23 26" src="https://user-images.githubusercontent.com/25318659/181005123-1b53a710-963e-4b47-8e64-e3b59189111b.png">

</td>

</tr>
</tbody>
</table>
</details>

## Test plan
- `sg start`
- enable `extensionsAsCoreFeatures` experimental feature
- ensure that git-extras extension is not loaded/activated
- ensure git blame works as before on file, commit and compare pages



## App preview:

- [Web](https://sg-web-taras-yemets-migrate-git-extras.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-iasindyvbl.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
